Sources/SyntaxKit/RegularExpression.swift

```diff
@@ -0,0 +1,266 @@
+//
+//  RegularExpression.swift
+//  SyntaxKit
+//
+//  Created by Zheng Wu on 2021/2/19.
+//  Copyright © 2021 Zheng Wu. All rights reserved.
+//
+
+import Foundation
+
+internal extension String {
```

Comment:
Why are we using CStrings here everywhere?

Reply:
Maybe it's faster? I have no better idea.

Reply:
I am making my progress to import Oniguruma and replace this one.

Reply:
CStrings should not be used, unless you are interoperating with C.

Reply:
About the other problem, I would maybe do a test with NSRegularExpression vs Oniguruma to see where they behave differently. Obviously it would be better to keep using Oniguruma.